servers/rendering/rendering_method.h

```diff
@@ -347,7 +347,7 @@ class RenderingMethod {
 		int info[RS::VIEWPORT_RENDER_INFO_TYPE_MAX][RS::VIEWPORT_RENDER_INFO_MAX] = {};
 	};
 
-	virtual void render_camera(const Ref<RenderSceneBuffers> &p_render_buffers, RID p_camera, RID p_scenario, RID p_viewport, Size2 p_viewport_size, uint32_t p_jitter_phase_count, float p_mesh_lod_threshold, RID p_shadow_atlas, Ref<XRInterface> &p_xr_interface, RenderInfo *r_render_info = nullptr) = 0;
+	virtual void render_camera(const Ref<RenderSceneBuffers> &p_render_buffers, RID p_camera, RID p_scenario, RID p_viewport, Size2 p_viewport_size, uint32_t p_jitter_phase_count, float p_mesh_lod_threshold, RID p_shadow_atlas, Ref<RefCounted> &p_xr_interface, RenderInfo *r_render_info = nullptr) = 0;
```

Comment:
The argument becoming `Ref<RefCounted>` in _all cases_, rather than `Ref<XRInterface>` is not ideal. Could we do something with macros so this switch only happens when XR is disabled? Like:

```c++
#ifdef XR_DISABLED
#define XRInterface RefCounted
#endif
```